okhttp-client/src/main/scala/org/http4s/client/okhttp/OkHttp.scala

```
@@ -0,0 +1,140 @@
+        ec.execute(new Runnable { override def run(): Unit = cb(Left(e)) })
+
+      override def onResponse(call: Call, response: OKResponse): Unit = {
+        val protocol = response.protocol() match {
+          case Protocol.HTTP_2 => HttpVersion.`HTTP/2.0`
+          case Protocol.HTTP_1_1 => HttpVersion.`HTTP/1.1`
+          case Protocol.HTTP_1_0 => HttpVersion.`HTTP/1.0`
+          case _ => HttpVersion.`HTTP/1.1`
+        }
+        val bodyStream = response.body.byteStream()
+        val dr = new DisposableResponse[F](
+          Response[F](headers = getHeaders(response), httpVersion = protocol)
+            .withStatus(
+              Status
+                .fromInt(response.code())
+                .getOrElse(Status.apply(response.code())))
+            .withBodyStream(
+              readInputStream(F.pure(bodyStream), chunkSize = 1024, closeAfterUse = true)),
+          F.delay({ bodyStream.close(); () })
+        )
+        ec.execute(new Runnable {
+          override def run(): Unit = cb(Right(dr))
+        })
+      }
+    }
+
+  private def getHeaders(response: OKResponse): Headers =
+    Headers(response.headers().names().asScala.toList.flatMap { v =>
+      response.headers().values(v).asScala.map(Header(v, _))
+    })
+
+  private def toOkHttpRequest[F[_]](req: Request[F])(implicit F: Effect[F]): OKRequest = {
+    val body = req match {
+      case _ if req.isChunked || req.contentLength.isDefined =>
+        new RequestBody {
+          override def contentType(): OKMediaType =
+            req.contentType.map(c => OKMediaType.parse(c.toString())).orNull
+
+          override def writeTo(sink: BufferedSink): Unit =
+            req.body.chunks
+              .map(_.toArray)
+              .to(Sink { b: Array[Byte] =>
+                F.delay {
+                  sink.write(b); ()
+                }
+              })
+              .compile
+              .drain
+              .runAsync(_ => IO.unit)
```

This swallows errors.  We should add something like cats-effect's `Logger.reportFailure`.